Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -396,25 +401,39 @@ WarpX::computePhiCartesian (const amrex::Vector<std::unique_ptr<amrex::MultiFab>
             );
         }
     }
-
-    // set the boundary potential values if needed
     setPhiBC(phi, dirichlet_flag, phi_bc_values_lo, phi_bc_values_hi);
 
     // Define the linear operator (Poisson operator)
     MLNodeTensorLaplacian linop( Geom(), boxArray(), DistributionMap() );
 
     // Set the value of beta
     amrex::Array<amrex::Real,AMREX_SPACEDIM> beta_solver =
-#if (AMREX_SPACEDIM==2)
+#   if (AMREX_SPACEDIM==2)
         {{ beta[0], beta[2] }};  // beta_x and beta_z
-#else
+#   else
         {{ beta[0], beta[1], beta[2] }};
-#endif
+#   endif
     linop.setBeta( beta_solver );
 
     // Solve the Poisson equation
     linop.setDomainBC( lobc, hibc );
 
+#else
+
+    // With embedded boundary: extract EB info
+    LPInfo info;
+    Vector<EBFArrayBoxFactory const*> eb_factory;
+    eb_factory.resize(max_level+1);
+    for (int lev = 0; lev <= max_level; ++lev) {
+      eb_factory[lev] = &WarpX::fieldEBFactory(lev);
+    }
+    MLEBNodeFDLaplacian linop( Geom(), boxArray(), dmap, info, eb_factory);
+
+    // TODO: Modify this
+    linop.setSigma({AMREX_D_DECL(1.0, 1.0, 1.0)});
+    linop.setEBDirichlet(0.);
```

Comment:
It would also be very useful if we could have user specified (and time dependent) potentials on the embedded boundary. Following the strategy for the domain boundaries this can be done via
```suggestion
    // get the EB potential
    std::string potential_eb_str = "0";
    ParmParse pp_embedded_boundary("eb2");
    pp_embedded_boundary.query("potential", potential_eb_str);
    auto parser_eb = makeParser(potential_eb_str, {"t"});
    linop.setEBDirichlet(parser_eb.eval(gett_new(0)));
```